lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/LibertyUtils.java

```diff
@@ -253,21 +253,18 @@ public static String getVersion(String serverXMLUri) {
      * Return temp directory to store generated feature lists and schema. Creates
      * temp directory if it does not exist.
      * 
-     * @param folder WorkspaceFolderURI indicates where to create the temporary
-     *               directory
+     * @param libertyWorkspace LibertyWorkspace where the temporary directory is made
+     *               
      * @return temporary directory File object
      */
-    public static File getTempDir(String workspaceFolderURI) {
-        if (workspaceFolderURI == null) {
+    public static File getTempDir(LibertyWorkspace libertyWorkspace) {
+        if (libertyWorkspace == null) {
             return null;
         }
         try {
-            URI rootURI = new URI(workspaceFolderURI);
-            Path rootPath = Paths.get(rootURI);
-            File file = rootPath.toFile();
-            File libertyLSFolder = new File(file, ".libertyls");
+            File libertyLSFolder = new File(findFileInWorkspace(libertyWorkspace,Paths.get("liberty-plugin-config.xml")).getParent().toFile(), ".libertyls");
 
-            if (!libertyLSFolder.exists()) {
+            if (libertyLSFolder != null && !libertyLSFolder.exists()) {
```

Comment:
I think we should go back to the old default location if the plugin file is not found.

Reply:
Should I keep this null check?

Reply:
Yes, but if it is null use the old default location to create the `.libertyls` folder.